hazelcast/src/main/java/com/hazelcast/spi/impl/AbstractCompletableFuture.java

```diff
@@ -110,25 +165,71 @@ public V get() throws InterruptedException, ExecutionException {
         }
     }
 
+    @Override
+    public V get(long timeout, TimeUnit unit) throws InterruptedException, ExecutionException, TimeoutException {
+        final long deadlineNanoTime = System.nanoTime() + unit.toNanos(timeout);
```

Comment:
just a small one: It's easier to work in ms rather then ns. `Object.wait()` will convert it to ms anyway -> you can save potentially expensive modulo operation, but more importantly - it will simplify the code.